### ISSUE TITLE
Add "caption" and "captionpos" option

### ISSUE THREAD

**Issue by gremaudc** [gpoore/minted#74]
Hello, 

I think it will be cool to add a "caption" and "captionpos" option. Before using Minted, I used `\begin{lstlisting}`

and there is a option for set directly the caption. Example.

`%\begin{lstlisting}[language=bash, caption="Installation of Oracle Java JDK 6 and 7",  captionpos=b]
%~# apt-get install python-software-properties --yes
%~# add-apt-repository ppa:webupd8team/java
%[Enter]
%~# apt-get update
%~# apt-get install oracle-java7-installer --yes
%\end{lstlisting}`

**Comment by klmr** [comment]
minted has the `listing `environment to do that. From the documentation:

```tex
\begin{listing}[H]
    \mint{cl}/(car (cons 1 '(2)))/ 
    \caption{Example of a listing.}
    \label{lst:example}
\end{listing}
```

**Comment by gremaudc** [comment]
Yes I know but im not able to do this.  In your example you use \mint
command which is only available for single line.... How can I use this
option with the std command. \begin{minted}... ?

Le jeu. 23 avr. 2015 11:23, Konrad Rudolph <notifications@github.com> a
écrit :

**Comment by klmr** [comment]
No, you misunderstood (a) what `mint` does and (b) how the `listing` environment works. Do read the documentation, it’s all explained there.

**Comment by gpoore** [comment]
As klmr has noted, `minted` can already do what you want with the `listing` environment, although it takes a little more typing than the case with the `listings` package.

Currently, the `minted` environment only typesets code. All additional functionality, likes captions, labels, and floats, is handled separately. I will think about whether caption-related options should be incorporated into the `minted` environment, or perhaps whether there should be a separate environment that functions as a shorthand for the `minted` and `listing` environment features combined. If additional functionality is added, it will probably take a while.

You should take a look at the `tcblisting` environment provided by the [`tcolorbox` package](https://www.ctan.org/pkg/tcolorbox?lang=en), since when combined with `minted` it already provides most or all of what you want.

**Comment by skanskan** [comment]
How can I reduce the top and bottom blank space (margins) if I use 
```
\begin{listing}[H]
    \mint{cl}/(car (cons 1 '(2)))/ 
    \caption{Example of a listing.}
    \label{lst:example}
\end{listing}
```
I need to write many of this codes within normal text and don't like to have too much white space in between.

**Comment by muzimuzhi** [comment]
@skanskan 
The vertical white space you mentioned is the vertical skip between every float environment and around text, and this skip is globally controlled by `\intextsep`, see https://latexref.xyz/Floats.html#Floats for detailed doc and other sep controllers.

`listing` environment is just an instance of float environment created by `minted`, with the help of `float` or `newfloat` pacakge.

**Comment by skanskan** [comment]
Now I'm doing it like this:

```
\definecolor{bg}{rgb}{0.9,0.95,1}
\newenvironment{code}{\captionsetup{type=table}}{}
\SetupFloatingEnvironment{listing}{name=Tabla}


\begin{code}
\begin{minted}[fontsize=\footnotesize,baselinestretch=1.3, bgcolor=bg]{text}
   my  code
\end{minted}
\caption{\label{mycaption} Mycaption.}
\end{code}
```

If I write two consecutive minted blocks the spacing is OK, but the caption (below) is confusing because it's not clear if it pertains to the code above or to the code below it.
How can I place it closer to its code or even better inside the colored area?
